[WIP] Styleguide as part of the starter kit

thibaudcolas opened the pull request:
@laurarbunea creating a PR so we can start to review – I know it's a bit early for that but it will help 🙂.

samanthaksanders commented:
Yes, after our css session the other day I realised this is probably useful for the less css inclined on our team. People don't use any of the sass stuff when they don't know what it's doing (which mean our lovely naming conventions get rewritten sometimes). So a visual representation outlining the useful things included in the kit will probably help more than I imagined 👍

laurarbunea commented:
NOTE: for now I have copied the compiled CSS into fred from another project, I need to change this to add a gulp task or  to compile it then copy it into here.

thibaudcolas commented:
I've moved the styleguide to the `docs` subfolder, and added an `npm run styleguide` command that will automatically do the build (grabbing the compiled styles from `core/static/css/styles.css`. Also changed the styleguide template so the output is loaded directly, without the need for JS/jQuery.

In the future we will want to re-create the styleguide on every `gulp css`, and (for the starter kit) publish it to the `gh-pages` branch so we can go to `https://springload.github.io/frontend-starter-kit` and see it (or a link to it).

thibaudcolas commented:
This is now live at https://springload.github.io/frontend-starter-kit/styleguide/

thibaudcolas commented on core/static_src/sass/components/_components.avatar.scss:
````diff
@@ -0,0 +1,42 @@
+<img class="avatar avatar--large" src="https://placekitten.com/150/150">
+<h5>2. avatar--small</h5>
+<img class="avatar avatar--small" src="https://placekitten.com/150/150">
+```
````
- For the titles, should we do the numbering in CSS (with a counter) so we can change the order of the snippets without having to update it?
- Should the titles be done in Markdown rather than HTML? It's a bit confusing to have the "labels" of the snippets be so close to the snippet.

This would be come:

------------------

To modify avatar you can add the modifiers listed below:

##### Base

```html
<img class="avatar" src="https://placekitten.com/150/150">
```

##### `.avatar--large`

```html
<img class="avatar avatar--large" src="https://placekitten.com/150/150">
```

##### `.avatar--small`

```html
<img class="avatar avatar--small" src="https://placekitten.com/150/150">
```

thibaudcolas commented on core/static_src/sass/components/_components.icon-text.scss:
```diff
@@ -1,16 +1,6 @@
-//     <svg>
-//         <use xlink:href="#arrow-next">
-//     </svg>
-// </span>
```
I quite like the idea of having the HTML snippet at the top – why did you choose to put it at the bottom @laurarbunea? Not sure which is best now.

laurarbunea replied:
sorry totally missed all these notes, I was subscribed to far too many repos. I had them at the bottom as I thought it woudl be more beneficial to see the code first but I think you are right and should be at the top !

thibaudcolas commented on core/static_src/sass/components/_components.modals.scss:
```diff
@@ -141,29 +122,64 @@ $color-modal-content-bg: #fff;
+
+Requires '_generic.keyframes.scss';
+
+As of 2015, the vertical-align: middle table is still the best cross-browser
```
We shouldn't date our code like that.

thibaudcolas commented on core/static_src/sass/components/_components.rich-text.scss:
```diff
@@ -1,3 +1,7 @@
+// Rich text
+// ============================================================================
+
 .rich-text {
```
Should this be a trump? It's not like it's a pattern we want to reuse – it's just overrides for our fav CMS isn't it?

thibaudcolas commented on core/static_src/sass/objects/_objects.grid.float.scss:
```diff
@@ -1,6 +1,6 @@
-// Float grid
-// -----------------------------------------------------------------------------
+// ============================================================================
+// Grid float
```
This should be Float grid – like Inline grid,  Justified grid.

thibaudcolas commented on core/static_src/sass/objects/_objects.list.unstyled.scss:
```diff
@@ -0,0 +1,24 @@
+// ============================================================================
+// Inline lists
```
Should be unstyled.

thibaudcolas commented on core/static_src/sass/components/_components.page.header.scss:
```diff
@@ -1,24 +1,44 @@
+
+
+/* SG
+# Components/page.header
```
General comment – with this, the SCSS comment at the top of the file, and the filename, we replicate the same info three times. Could we get rid of one of them?

thibaudcolas commented on core/static_src/sass/theming/_theming.scss:
```diff
@@ -62,3 +68,47 @@ $themes: (
+                <p class='t-pink'>t-pink</p>
+            </div>
+        </div>
+    </div>
```
I don't think this example requires using the grid classes - or does it?

thibaudcolas commented on core/static_src/sass/tools/_tools.breakpoints.scss:
```diff
@@ -43,3 +44,64 @@ $breakpoint-xlarge: 1280px;
+.breakpoint-on-medium,
+.breakpoint-on-large,
+.breakpoint-on-xlarge,
+{ padding:1em; }
```
This should be within the `styleguide.scss` file, not the breakpoints file.

thibaudcolas commented on core/static_src/sass/utilities/_utilities.debug.scss:
```diff
@@ -0,0 +1,33 @@
+// Define styles for errors and warns that we can reuse throughout this file.
```
This file isn't used anywhere, should we get rid of it or use it?

thibaudcolas commented on core/static_src/sass/utilities/_utilities.debug.scss:
```diff
@@ -0,0 +1,33 @@
+ * `#0` if we have to point to `#` at all.
+ */
+[href="#"] {
+  @include debug-warn();
```
I don't think that's a good idea – don't see what's the benefit of pointing to `#0`.

thibaudcolas commented on docs/styleguide/template.html:
```diff
@@ -0,0 +1,220 @@
+</head>
+<body style='background-color:#fff; height:100%'>
+
+    <div class='[ wrapper ] t-plain header'  role="contentinfo" >
```
Are we keeping those square brackets?

thibaudcolas commented on docs/styleguide/template.html:
```diff
@@ -0,0 +1,220 @@
+        <div class='[ wrapper wrapper--narrow ]' style=''>
+            <div class='[ section gutters ]'>
+                <div class='u-text-center '>
+                    <img src='../styleguide/fred-logo.png' alt='Fred the friendly front end' class='u-mb'>
```
I think this should be `alt=''`.

thibaudcolas commented on docs/styleguide/template.html:
```diff
@@ -0,0 +1,220 @@
+                    <li><strong>UTILITIES</strong> – utilities and helper classes with ability to override anything which goes before in the triangle, eg. hide helper class</li>
+                    <li><strong>THEMING</strong> – website themes</li>
+                </ol>
+                 <!--<p class='u-text-center'><img src='../styleguide/itcss-triangle.png' alt='Inverted triangle'/></p>-->
```
We need to find an image that we have the rights to use, or make our own.

thibaudcolas commented:
I'll look into merging this today 🚀.

thibaudcolas commented:
Merging this now with a few changes and cleanup, we can spend more time on this later 🚀

laurarbunea commented on core/static_src/sass/_components.scss:
```diff
@@ -26,9 +27,15 @@
+@import 'components/components.table.scrollable';
+@import 'components/components.table.stacked';
+@import 'components/components.table.striped';
+@import 'components/components.pagination';
```
I think so pagination styles are always helpful and its good to have one as a starting point. We do need to address the issue at somepoint about a library vs basic starting kit for projects so maybe once we have this issue addressed we can add more to the library rather than the starter kit... or we just have the library and people pull out the styles by copy and paste for the projects.  I have talked to the designers and they would to see our library of re usable components we have done. I might add this as a issue so we can talk about it more with the team :)
